### ISSUE TITLE
Documentation generation

### ISSUE THREAD

**Issue by amerkoleci** [SharpGenTools/SharpGenTools#116]
Hi,

I know there is https://www.nuget.org/packages/SharpGen.Doc.Msdn.Tasks/ but I would like to know how to generate documentation for my SharpWindows libraries?

Are there any other better way?
Thanks

**Comment by jkoritzinsky** [comment]
At the moment there isn't really a good option since MSDN has been shut down. You could write your own doc provider to pull from docs.microsoft.com (which is basically the same idea as the MSDN doc provider) or write an XML doc comments file and add it to your build (which is what I would suggest). I'm not currently planning on creating and maintaining a new external docs provider at the moment.

**Comment by amerkoleci** [comment]
Are there any tutorial on how to write XML doc comments file?

**Comment by jkoritzinsky** [comment]
I'm not sure if I ever got around to writing that doc. I'll put that on my list of things to do!

**Comment by Axiverse** [comment]
I'm not sure what the id looks for. The docs seem to suggest a C++ name, but I've tried the C# full name as well. It doesn't seem to bind. E.g. the following doesn't bind any comments.

```
<?xml version="1.0" encoding="utf-8" ?>
<comments>
  <comment id="ID3D11Device">
    <summary>Hello World ID3D11Device</summary>
  </comment>
  <comment id="D3D11_CPU_ACCESS_FLAG">
    <summary>Hello World D3D11_CPU_ACCESS_FLAG</summary>
  </comment>
</comments>
```

With `<SharpGenExternalDocs Include="Documentation.xml" />` added to the project file right under `<SharpGenMapping Include="Mappings.xml" />`

**Comment by Axiverse** [comment]
The include doc tag needs to be closed otherwise it's malformed. #170 

There is still an issue where we link the methods but no comment is able to be generated for the enclosing classes. Thus interfaces like ID3D11Device have no comment binding, only the contained methods.

**Comment by Axiverse** [comment]
Yeah, I'm unable to get class/interface level comments even if it's explicitly mapped.

**Comment by Axiverse** [comment]
Modified InterfaceCodeGenerator to include documentation trivia when generating classes and interfaces.

**Comment by AronRubin** [comment]
My own solution is to write a dummy c++ file that includes all of the desired headers. I then compile that with
`cl /doc dummy.cpp`
Then I apply a XSL to wrestle it to the form read by SharpGenTools. It was a bit of a pain to write the XSL because I had to deal with doxygen style comments and SharpGenTools needs namespaces stripped.
